test/verify/check-shell-multi-machine-key

```diff
@@ -239,15 +239,14 @@ class TestMultiMachineKeyAuth(MachineCase):
         m2 = self.machine2
 
         # upload id_ed25519 (password: locked)
-        m1.execute("mkdir -p /home/admin/.ssh")
-        m1.upload(["verify/files/ssh/id_ed25519", "verify/files/ssh/id_ed25519.pub",
-                   "verify/files/ssh/id_rsa", "verify/files/ssh/id_rsa.pub"],
-                  "/home/admin/.ssh/")
         m1.write("/home/admin/.ssh/config", """
 Host 10.111.113.2
     User user
     IdentityFile /home/admin/.ssh/id_ed25519
 """)
+        m1.upload(["verify/files/ssh/id_ed25519", "verify/files/ssh/id_ed25519.pub",
```

Comment:
This reordering is begging for `mkdir -p` to be added to `Machine.upload()` as well.

Reply:
Yes, I thought the same. It's on my TODO list now, next to the dedent() thing.